src/main/java/com/pertos/bring/context/ApplicationContext.java

```diff
@@ -0,0 +1,15 @@
+package com.pertos.bring.context;
+
+import com.pertos.bring.exception.NoSuchBeanException;
+import com.pertos.bring.exception.NoUniqueBeanException;
+
+import java.util.Map;
+
+public interface ApplicationContext {
+
+    <T> T getBean(Class<T> beanType) throws NoSuchBeanException, NoUniqueBeanException;
```

Comment:
once exceptions defined here shouldn't they be checked ?

Reply:
I'm not sure it is a good idea to force our client to catch checked exceptions. Here is an example how Spring's ApplicationContext works:
`ApplicationContext context = SpringApplication.run(MockServerApplication.class, args);
 Object notExistedClass = context.getBean("notExistedClass");`
Application will be terminated with exception in logs:
`Exception in thread "main" org.springframework.beans.factory.NoSuchBeanDefinitionException: No bean named 'notExistedClass' available`